ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -483,29 +498,11 @@ loadSessionWithOptions recorder SessionLoadingOptions{..} dir = do
     -- combined with the components in the old HscEnv into a new HscEnv
     -- which contains the union.
     let packageSetup :: (Maybe FilePath, NormalizedFilePath, ComponentOptions, FilePath)
-                     -> IO (HscEnv, ComponentInfo, [ComponentInfo])
+                     -> IO ([ComponentInfo], [ComponentInfo])
         packageSetup (hieYaml, cfp, opts, libDir) = do
           -- Parse DynFlags for the newly discovered component
           hscEnv <- emptyHscEnv ideNc libDir
```

Comment:
Out of curiosity, do we still need to reuse the NameCache in all situations?
In theory, I don't think so for mhu. Maybe we should keep the comment about the NameCache for GHC 9.2 and 9.4?

Reply:
I think we should still reuse the NameCache.